controllers/apps/operations/backup.go

```diff
@@ -0,0 +1,179 @@
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package operations
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	dataprotectionv1alpha1 "github.com/apecloud/kubeblocks/apis/dataprotection/v1alpha1"
+	"github.com/apecloud/kubeblocks/internal/constant"
+	intctrlutil "github.com/apecloud/kubeblocks/internal/controllerutil"
+)
+
+type BackupOpsHandler struct{}
+
+var _ OpsHandler = BackupOpsHandler{}
+
+func init() {
+	// ToClusterPhase is not defined, because 'backup' does not affect the cluster phase.
+	backupBehaviour := OpsBehaviour{
+		FromClusterPhases:                  appsv1alpha1.GetClusterUpRunningPhases(),
+		OpsHandler:                         BackupOpsHandler{},
+		ProcessingReasonInClusterCondition: ProcessingReasonBackup,
+	}
+
+	opsMgr := GetOpsManager()
+	opsMgr.RegisterOps(appsv1alpha1.BackupType, backupBehaviour)
+}
+
+// ActionStartedCondition the started condition when handling the backup request.
+func (b BackupOpsHandler) ActionStartedCondition(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) (*metav1.Condition, error) {
+	return appsv1alpha1.NewBackupCondition(opsRes.OpsRequest), nil
+}
+
+// Action implements the backup action.
+// It will create a backup resource for cluster.
+func (b BackupOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
+	opsRequest := opsRes.OpsRequest
+	cluster := opsRes.Cluster
+
+	// create backup
+	if backup, err := buildBackup(reqCtx, cli, opsRequest, cluster); err != nil {
+		return err
+	} else {
+		return cli.Create(reqCtx.Ctx, backup)
+	}
+}
+
+// ReconcileAction implements the backup reconcile action.
+// It will check the backup status and update the OpsRequest status.
+// If the backup is completed, it will return OpsSuccess
+// If the backup is failed, it will return OpsFailed
+func (b BackupOpsHandler) ReconcileAction(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) (appsv1alpha1.OpsPhase, time.Duration, error) {
+	opsRequest := opsRes.OpsRequest
+	cluster := opsRes.Cluster
+
+	// get backup
+	backups := &dataprotectionv1alpha1.BackupList{}
+	if err := cli.List(reqCtx.Ctx, backups, client.InNamespace(cluster.Namespace), client.MatchingLabels(getBackupLabels(cluster.Name, opsRequest.Name))); err != nil {
+		return appsv1alpha1.OpsFailedPhase, 0, err
+	}
+
+	if len(backups.Items) == 0 {
+		return appsv1alpha1.OpsFailedPhase, 0, fmt.Errorf("backup not found")
+	}
+	// check backup status
+	phase := backups.Items[0].Status.Phase
+	if phase == dataprotectionv1alpha1.BackupCompleted {
+		return appsv1alpha1.OpsSucceedPhase, 0, nil
+	} else if phase == dataprotectionv1alpha1.BackupFailed {
+		return appsv1alpha1.OpsFailedPhase, 0, fmt.Errorf("backup failed")
+	}
+	return appsv1alpha1.OpsRunningPhase, time.Second, nil
```

Comment:
watch the backup by opsRequest's label in opsRequest controller.

Reply:
I watch backup by these label, which contains the opsRequest's name and type.
https://github.com/apecloud/kubeblocks/blob/ea6bfc6b83690ee9863358cc030b4f2938c601d4/controllers/apps/operations/backup.go#L174-L180